libs/db/boltdb.go

```diff
@@ -0,0 +1,253 @@
+package db
```

Comment:
Not blocking this PR but shouldn't we create a package per implementation (bolt, golevel, mem, etc)?

Reply:
Yes to ^. We should also use conditional compilation to only include deps & code when X db is requested (`// +build boltdb`).

Reply:
https://github.com/tendermint/tendermint/issues/3611